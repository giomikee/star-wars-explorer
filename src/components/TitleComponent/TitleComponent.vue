<template>
  <v-container class="d-flex justify-center align-center ga-1">
    <v-img
      alt="Star wars logo"
      height="50"
      max-width="50"
      src="/star-wars-logo.svg"
    />
    <h1 class="d-md-none d-sm-block text-h4">Star Wars Explorer</h1>
    <h1 class="d-none d-md-block text-h2">Star Wars Explorer</h1>
  </v-container>
</template>

<script lang="ts">
  export default { name: 'TitleComponent' }
</script>
