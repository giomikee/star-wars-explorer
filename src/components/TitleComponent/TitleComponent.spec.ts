import { shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import TitleComponent from './TitleComponent.vue'

const createWrapper = () => shallowMount(TitleComponent)

describe('Given TitleComponent component', () => {
  describe('When component is rendered', () => {
    it('Then it should match snapshot', () => {
      const { element } = createWrapper()

      expect(element).toMatchSnapshot()
    })
  })
})
