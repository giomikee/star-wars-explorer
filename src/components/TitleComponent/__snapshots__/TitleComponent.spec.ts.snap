// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given TitleComponent component > When component is rendered > Then it should match snapshot 1`] = `
<div
  class="v-container v-locale--is-ltr d-flex justify-center align-center ga-1"
>
  <v-img-stub
    absolute="false"
    alt="Star wars logo"
    cover="false"
    eager="false"
    height="50"
    inline="false"
    maxwidth="50"
    options="[object Object]"
    src="data:image/svg+xml,%3c!DOCTYPE%20svg%20PUBLIC%20'-//W3C//DTD%20SVG%201.1//EN'%20'http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd'%3e%3c!--%20Uploaded%20to:%20SVG%20Repo,%20www.svgrepo.com,%20Transformed%20by:%20SVG%20Repo%20Mixer%20Tools%20--%3e%3csvg%20width='800px'%20height='800px'%20viewBox='0%200%20192.756%20192.756'%20xmlns='http://www.w3.org/2000/svg'%20fill='%23ffffff'%3e%3cg%20id='SVGRepo_bgCarrier'%20stroke-width='0'/%3e%3cg%20id='SVGRepo_tracerCarrier'%20stroke-linecap='round'%20stroke-linejoin='round'/%3e%3cg%20id='SVGRepo_iconCarrier'%3e%3cg%20fill-rule='evenodd'%20clip-rule='evenodd'%3e%3cpath%20fill='%2300000000'%20d='M0%200h192.756v192.756H0V0z'/%3e%3cpath%20d='M5.669%2081.215v12.65h37.003c4.301%200%209.796-3.977%209.796-10.556%200-2.646%201.012-4.372-2.098-7.872l-4.733-5.608c-2.712-2.53.324-2.53%202.602-2.53h15.623v26.566h12.39V67.299h16.699V55.922H38.877c-6.579%200-9.796%206.317-9.615%209.606.182%203.289.787%207.427%206.254%2012.398%204.987%204.533-2.469%203.289-3.218%203.289H5.669zM120.348%2055.922H100.36L89.155%2093.866h12.47l2.023-5.313h13.156l1.953%205.313h12.215l-10.624-37.944zm-13.916%2023.522l4.301-13.916%204.049%2013.916h-8.35zM170.443%2081.215c-4.807%200-4.807-1.771-4.807-1.771%204.119%200%207.771-6.001%207.771-12.145s-6-11.377-10.809-11.377h-26.891v37.944h13.664v-12.65s5.818%206.831%208.854%209.614c3.037%202.783%203.289%203.036%207.41%203.036h21.449v-12.65c.002-.001-11.834-.001-16.641-.001zm-12.398-8.855h-8.672v-6.832h8.672c3.976%200%204.664%206.832%200%206.832zM5.669%2098.672h13.979l3.542%2012.652%203.289-12.652h14.675l3.795%2012.652%203.796-12.652h12.144l-11.133%2037.953H38.624l-4.878-17.965-5.496%2017.965H16.865L5.669%2098.672zM89.578%2098.891H69.59l-11.204%2037.943h12.469l2.024-5.312h13.157l1.953%205.312h12.216L89.578%2098.891zm-13.915%2023.521l4.301-13.916%204.048%2013.916h-8.349zM170.695%20110.059c-2.275%200-4.756.266-2.043%202.795l4.734%205.609c3.109%203.5%203.059%204.959%203.059%207.607%200%206.578-6.508%2010.555-10.809%2010.555l-29.896.201c-4.119%200-4.371-.252-7.408-3.035-3.035-2.783-8.855-9.615-8.855-9.615v12.65h-13.662V98.883h26.891c4.807%200%2010.809%205.234%2010.809%2011.377%200%206.145-3.652%2012.145-7.773%2012.145%200%200%201.812%201.822%204.848%201.822%203.037%200%2014.727.012%2014.727.012.748%200%208.203%201.244%203.217-3.289-5.467-4.971-6.072-9.107-6.254-12.396s2.662-9.881%209.238-9.881h25.57v11.387h-16.393v-.001zm-42.545%205.261h-8.674v-6.832h8.674c3.977%200%204.664%206.832%200%206.832z'/%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
    tile="false"
    transition="fade-transition"
  />
  <h1
    class="d-md-none d-sm-block text-h4"
  >
    Star Wars Explorer
  </h1>
  <h1
    class="d-none d-md-block text-h2"
  >
    Star Wars Explorer
  </h1>
</div>
`;
